00_ES6/00_start/src/common/sessionProvider.jsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+export const SessionContext = React.createContext()
+
+export class SessionProvider extends React.Component {
+  state = {
+     room : '',
+     nickname: '',
+     setChatSessionInfo: (nickname, room) => {
```

Comment:
why include this method on state?

Reply:
Context, to be able to update context.

Take from

Using actions and modifying data in the React Context
Unless you are working on a very basic app you will need a way to update/change the data that comes from the React Context. The minimal example can be a button that will increment the number from our data.

```javascript
class AppProvider extends Component {
 state = {
    number : 10,
    inc: () => {
      this.setState({number: this.state.number + 1})
    }
  }
  //...
}
```

````javascript
const Blue = () => (
  <div className="blue">
    <AppContext.Consumer>
        {(context) => <button onClick={context.inc}>INC</button>}
      </AppContext.Consumer>
    <Green />
  </div>
)
```

https://hackernoon.com/how-to-use-the-new-react-context-api-fce011e7d87

Reply:
I suppose it's a short hand of:

```
export class SessionProvider extends React.Component {
  state = {
     room : '',
     nickname: '',
   }

    setChatSessionInfo = (nickname, room) => {
        this.setState({nickname, room})
     }

    render() {
    return (
       <SessionContext.Provider
          value={{
              nickname: this.state.nickname,
              room: this.state.room,
              setChatSessionInfo: this.setChatSessionInfo,
           }}
        >
            {this.props.children}
       </SessionContext.Provider>
    );
   }
 } 
```

It's weird for me see methods on React state

Reply:
Yes, the thin is in the consumer you may want to update a value, that's why is on state, you are right sounded strange to me, but on the other hand it may have sense, you may want to update a value in many places